src/board_controller/openbci/ganglion_bglib/callbacks.cpp

```diff
@@ -68,8 +71,18 @@ void ble_evt_gap_scan_response (const struct ble_msg_gap_scan_response_evt_t *ms
 
     if (name_found_in_response)
     {
-        // strcasestr is unavailable for windows
-        if (strstr (name, "anglion") != NULL)
+        // Determine the firmware version from the advertised name
+        if (msg->data.data[0] == 20 && msg->data.data[13] == '3')
+        {
+            GanglionLib::firmware = 3;
+        }
+
+        if (memcmp (msg->sender.addr, (uint8 *)GanglionLib::connect_addr.addr, sizeof (bd_addr)) ==
+            0)
+        {
+            GanglionLib::exit_code = (int)GanglionLib::STATUS_OK;
```

Comment:
why is it added?

Reply:
The OpenBCI GUI uses the serial_number brainflow parameter to pass the advertised name of the expected device. This is important because if two Ganglions are active, the user will pick one or the other device from the GUI. Brainflow needs to know which device to connect to. This could probably be done via a MAC address, but since this is the way it's currently being done, I haven't changed it.

Reply:
ok, got it. before we were doing filtering by mac address if I remember correctly and gui passed mac addressses instead names(user still picked the name)